src/main/java/com/lothrazar/storagenetwork/util/RequestBatch.java

```diff
@@ -0,0 +1,26 @@
+package com.lothrazar.storagenetwork.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.lothrazar.storagenetwork.api.IConnectableLink;
+
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+
+public class RequestBatch extends Batch<Request> {
+    public void extractStacks(IConnectableLink providerStorage, Integer slot, Item item) {
+        List<Request> requests = get(item);
+        List<Request> remainingRequests = new ArrayList<Request>();
+        for (Request request : requests) {
+            if (!request.insertStack(providerStorage, slot)) {
+                remainingRequests.add(request);
```

Comment:
There could be a better way to track the remaining requests, that creating a new list for them

Reply:
Gonna keep it this way for the time beeing. editing the list while in the loop throws an error.